src/Service/ActivityPub/ApHttpClient.php

```diff
@@ -168,64 +199,72 @@ private function getActorCacheKey(string $apProfileId): string
-                        $magazine->apTimeoutAt = new \DateTime();
-                        $this->magazineRepository->save($magazine, true);
-                    }
-
-                    $msg = "AP Get fail: $apProfileId, ex: ".\get_class($e).": {$e->getMessage()}";
-                    if (null !== $response) {
-                        $msg .= ', '.$response->getContent(false);
-                    }
-                    throw new InvalidApPostException($msg);
-                }
+        $key = $this->getActorCacheKey($apProfileId);
+        if ($this->cache->hasItem($key)) {
+            /** @var CacheItem $item */
+            $item = $this->cache->getItem($key);
+            $resp = $item->get();
 
-                $item->expiresAt(new \DateTime('+1 hour'));
+            return $resp ? json_decode($resp, true) : null;
+        }
 
-                if (404 === $response->getStatusCode()) {
-                    // treat a 404 error the same as a tombstone, since we think there was an actor, but it isn't there anymore
-                    return json_encode($this->tombstoneFactory->create($apProfileId));
-                }
+        $resp = $this->getActorObjectImpl($apProfileId);
 
-                // Return the content.
-                // Pass the 'false' option to getContent so it doesn't throw errors on "non-OK" respones (eg. 410 status codes).
-                return $response->getContent(false);
-            }
-        );
+        /** @var CacheItem $item */
+        $item = $this->cache->getItem($key);
+        $item->expiresAt(new \DateTime('+1 hour'));
+        $item->set($resp);
+        $this->cache->save($item);
 
         return $resp ? json_decode($resp, true) : null;
     }
 
+    private function getActorObjectImpl(string $apProfileId): ?string
+    {
+        $this->logger->debug("ApHttpClient:getActorObject:url: $apProfileId");
+        $response = null;
+        try {
+            // Set-up request
+            $client = new CurlHttpClient();
+            $response = $client->request('GET', $apProfileId, [
+                'max_duration' => self::MAX_DURATION,
+                'timeout' => self::TIMEOUT,
+                'headers' => $this->getInstanceHeaders($apProfileId, null, 'get', ApRequestType::ActivityPub),
+            ]);
+            // If 4xx error response, try to find the actor locally
+            if (str_starts_with((string) $response->getStatusCode(), '4')) {
+                if ($user = $this->userRepository->findOneByApProfileId($apProfileId)) {
+                    $user->apDeletedAt = new \DateTime();
+                    $this->userRepository->save($user, true);
+                }
+                if ($magazine = $this->magazineRepository->findOneByApProfileId($apProfileId)) {
+                    $magazine->apDeletedAt = new \DateTime();
+                    $this->magazineRepository->save($magazine, true);
+                }
+            }
+        } catch (\Exception|TransportExceptionInterface $e) {
```

Comment:
Interesting I would though that `TransportExceptionInterface` was extending from Exception, but it isn't in Symfony. Instead TransportExceptionInterface-> ExceptionInterface -> Throwable.

I think this is also a bad design pattern by Symfony IMO.